edsl.md

```diff
@@ -159,58 +159,64 @@ where `F1 : F2 : F3 : F4` is the (two's complement) byte-array representation of
     rule #enc(#address( DATA )) => #buf(32, #getValue(#address( DATA )))
     rule #enc(#uint256( DATA )) => #buf(32, #getValue(#uint256( DATA )))
     rule #enc( #uint48( DATA )) => #buf(32, #getValue( #uint48( DATA )))
-    rule #enc( #uint16( DATA )) => #buf(32, #getValue( #uint16( DATA )))    
+    rule #enc( #uint16( DATA )) => #buf(32, #getValue( #uint16( DATA )))
     rule #enc(  #uint8( DATA )) => #buf(32, #getValue(  #uint8( DATA )))
     rule #enc( #int256( DATA )) => #buf(32, #getValue( #int256( DATA )))
     rule #enc( #int128( DATA )) => #buf(32, #getValue( #int128( DATA )))
     rule #enc(#bytes32( DATA )) => #buf(32, #getValue(#bytes32( DATA )))
     rule #enc(   #bool( DATA )) => #buf(32, #getValue(   #bool( DATA )))
 
     // dynamic Type
-    rule #enc(        #bytes(WS)) => #encBytes(#sizeWordStack(WS), WS)
+    rule #enc(        #bytes(BS)) => #encBytes(#sizeByteArray(BS), BS)
     rule #enc(#array(_, N, DATA)) => #enc(#uint256(N)) ++ #encodeArgs(DATA)
     rule #enc(      #string(STR)) => #enc(#bytes(#parseByteStackRaw(STR)))
 
-    syntax ByteArray ::= #encBytes ( Int , WordStack ) [function]
+    syntax ByteArray ::= #encBytes ( Int , ByteArray ) [function]
  // -------------------------------------------------------------
-    rule #encBytes(N, WS) => #enc(#uint256(N)) ++ WS ++ #buf(#ceil32(N) -Int N, 0)
+    rule #encBytes(N, BS) => #enc(#uint256(N)) ++ BS ++ #buf(#ceil32(N) -Int N, 0)
 
     //Byte array buffer. Lemmas defined in evm-data-symbolic.k
     // SIZE, DATA // left zero padding
     syntax ByteArray ::= #buf ( Int , Int ) [function, smtlib(buf)]
  // ---------------------------------------------------------------
+    rule #buf(SIZE, DATA) => #padToWidth(SIZE, #asByteStack(DATA))
+      requires #range(0 <= DATA < (2 ^Int (SIZE *Int 8))) [concrete]
+
+    syntax ByteArray ::= #bufSeg ( ByteArray , Int , Int ) [function, smtlib(bufSeg)] // BUFFER, START, WIDTH
```

Comment:
@daejunpark I'm not sure if this should be put here or in verification.k. I put it here mainly because the haskell backend will fail to unparse terms if the definition is in verification.k

Reply:
It is fine to put these here, but the unparsing issue should be fixed. We wouldn't want to put all the syntax declarations here.

Reply:
I've removed it for now, since it's not needed to make this PR work.